Add regression test for fixed crasher

beccadax opened the pull request:
This doesn’t crash in current master, but it did in Swift 4.1.2. Adding it to our fixed crashers to make sure it stays that way.

beccadax committed:
Add prior crasher

This test actually diagnoses the error correctly in current master, but it crashed in a previous version of Swift. Let’s make sure we don’t bring it back.

slavapestov commented on validation-test/compiler_crashers_2_fixed/28872-rdar45583135.swift:
```diff
@@ -0,0 +1,16 @@
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+
+// RUN: not %target-swift-frontend %s -emit-sil
```
Once we fix the real issue here, this code will actually type check (and the test can fail, so we can update it then)

beccadax committed:
Fling a light into the future

Just leaving a note for whoever needs to debug this test “breaking” by starting to work properly.

brentdax commented:
@swift-ci please test
